src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -43,8 +43,19 @@ public void TestEnvironmentProperty()
             environment.Remove("NewKey");
             Assert.Equal(CountItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // Exception thrown with invalid key
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
+            // The full .NET Framework uses CaseSensitiveStringDictionary, but we can't have an
+            // equivilent as we don't have access to the internal state of StringDictionary in .NET Core.
+            if (PlatformDetection.IsFullFramework)
```

Comment:
@hughbe I believe this behavior was ported like this by design. I think it makes the implementation easier and better to understand. This was ported in #13203 by myself.

cc: @stephentoub @danmosemsft any thoughts?

I think we should open an issue as netfx-port-consider to port this to behave as core using dictionary instead of hash table.